src/canvas-extractor/tests/graphql/extractor/test_extractor_datefilter.py

```diff
@@ -0,0 +1,40 @@
+# SPDX-License-Identifier: Apache-2.0
+# Licensed to the Ed-Fi Alliance under one or more agreements.
+# The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
+# See the LICENSE and NOTICES files in the project root for more information.
+
+import pytest
+
+from edfi_canvas_extractor.graphql.utils import validate_date
+
+
+def test_graphql_request(api):
+    # start = "2021-01-01"
+    # end = "2030-01-01"
+
+    courses = api["data"]["account"]["coursesConnection"]["nodes"][0]
+
+    for course in courses["term"]["coursesConnection"]["nodes"]:
+        if course["state"] not in ["available", "completed"]:
+            continue
+
+        start_term = courses["term"]["startAt"]
+        end_term = courses["term"]["endAt"]
+
+        assert start_term is None
+        assert end_term is None
+
+
+@pytest.mark.xfail
```

Comment:
COULD: Instead of marking this test (and the next one) as `expected to fail` , we could raise a `ValueError` exception in `validate_date()` and assert that error in these tests.